generic_config_updater/gu_common.py

```diff
@@ -63,13 +64,16 @@ def get_config_db_as_json(self):
         return config_db_json
 
     def _get_config_db_as_text(self):
-        # TODO: Getting configs from CLI is very slow, need to get it from sonic-cffgen directly
-        cmd = "show runningconfiguration all"
-        result = subprocess.Popen(cmd, shell=True, text=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if self.namespace is not None and self.namespace != multi_asic.DEFAULT_NAMESPACE:
+            cmd = ['sonic-cfggen', '-d', '--print-data', '-n', self.namespace]
+        else:
+            cmd = ['sonic-cfggen', '-d', '--print-data']
+
+        result = subprocess.Popen(cmd, shell=False, text=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         text, err = result.communicate()
         return_code = result.returncode
         if return_code: # non-zero means failure
-            raise GenericConfigUpdaterError(f"Failed to get running config, Return code: {return_code}, Error: {err}")
+            raise GenericConfigUpdaterError(f"Failed to get running config for namespace: {self.namespace}, Return code: {return_code}, Error: {err}")
```

Comment:
There are other preexisting exceptions raised not touched in this PR yet, such as 
   `raise GenericConfigUpdaterError("Attempting to call invalid method {} in module {}. Module must be generic_config_updater.field_operation_validators, and method must be a defined validator".format(method_name, module_name))` also in this file. Should we add namespace information to these other exception logs as well?